_rules/SC4-1-2-iframe-has-name.md

```diff
@@ -0,0 +1,149 @@
+
+## Assumptions
+
+- The rule assumes that the target `iframe` is used as an user interface component.
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [H64: Using the title attribute of the frame and iframe elements](http://www.w3.org/TR/WCAG20-TECHS/H64.html)
+- [Understanding Success Criterion 4.1.2 | Understanding WCAG 2.0](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html)
+- [Understanding Success Criterion 2.4.1 | Understanding WCAG 2.0](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-skip.html)
+
+## Test cases
+
+### Passed
+
+#### Pass example 1
+
+Usage of `title` attribute to describe the `iframe` content.
+
+```html
+<iframe title="List of Contributors" src="../test-assets/SC4-1-2-frame-doc.html">
+</iframe>
+```
+
+#### Pass example 2
+
+Usage of `aria-label` attribute to describe the `frame` content.
+
+```html
+<iframe aria-label="Advertisement of tours to Great Wall of China" src="../test-assets/SC4-1-2-frame-doc.html" >
+</iframe>
+```
+
+#### Pass example 3
+
+Usage of `aria-labelledby` attribute to describe the `iframe` content.
+
+```html
+<div id="frame-title-helper">Watch highlights of the Worldcup</div>
+<iframe aria-labelledby="frame-title-helper" src="../test-assets/SC4-1-2-frame-doc.html">
+</iframe>
+```
+
+### Failed
+
+#### Fail example 1
+
+Usage of `name` attribute to describe the `iframe` content is not valid.
+
+```html
+<iframe name="List of Contributors" src="../test-assets/SC4-1-2-frame-doc.html" >
+</iframe>
+```
+
+#### Fail example 2
+
+`iframe` with no `title`/`aria-label`/`aria-labelledby` attribute to describe content is not valid.
+
+```html
+<iframe src="../test-assets/SC4-1-2-frame-doc.html" >
+</iframe>
+```
+
+#### Fail example 3
+
+Empty `title` attribute is not valid.
+
+```html
+<iframe title="" src="../test-assets/SC4-1-2-frame-doc.html" >
+</iframe>
+```
+
+#### Fail example 4
+
+Empty `aria-label` attribute to describe the `frame` content is not valid.
+
+```html
+<iframe aria-label="" src="../test-assets/SC4-1-2-frame-doc.html" >
+</iframe>
+```
+
+#### Fail example 5
+
+Usage of non existing `aria-labelledby` reference element to describe the `iframe` content is not valid.
+
+```html
+<iframe aria-labelledby="does-not-exist" src="../test-assets/SC4-1-2-frame-doc.html">
+</iframe>
+```
+
+#### Fail example 6
+
+Usage of `alt` attribute to describe content is not valid.
+
+```html
+<iframe alt="List of Contributors" src="../test-assets/SC4-1-2-frame-doc.html">
+</iframe>
+```
+
+### Inapplicable
+
+#### Inapplicable example 1
+
+Does not apply to non `iframe` element.
+
+```html
+<button>take me somewhere</button>
+```
+
+#### Inapplicable example 2
+
+Usage of `aria-hidden="true"` does not expose `iframe` to [assistive technologies](#exposed-to-assistive-technologies).
+
+```html
+<iframe aria-hidden="true" src="../test-assets/SC4-1-2-frame-doc.html">
```

Comment:
aria-hidden elements are still available to assistive technologies.

Reply:
We have used `aria-hidden="true"` in other rules as an example to not expose to assistive technology. (Eg: button has name, inapplicable example 2).

I checked on Firefox accessibility tree, to ensure an element with `aria-hidden=true` is not shown in the tree, but I am not sure if assistive technology uses a similar tree to parse and read content.